src/iface-ethernet-ip.c

```diff
@@ -97,6 +97,34 @@ static int verify_tag_name_created(struct knot_data_item *data_item)
 	return rc;
 }
 
+static int put_tag_name_created(struct knot_data_item *data_item)
+{
+	struct knot_data_item *data_item_aux;
+	int i;
+	int rc = -1;
+
+	for (i = 0; i < thing_ethernet_ip.number_sensor; i++) {
+		int return_aux;
+
+		data_item_aux = l_hashmap_lookup(thing_ethernet_ip.data_items,
+						 L_INT_TO_PTR(i));
+		if (!data_item_aux)
+			rc = -EINVAL;
+
+		return_aux = strcmp(
+			data_item_aux->tag_name,
+			data_item->tag_name);
```

Comment:
Do we need to put a guard condition since the array is statically declared inside the structure?

Reply:
The data_item_aux receive its value from a l_hashmap_lookup and it may returns NULL, my concern is about trying to access a struct field in a NULL value.